docs/src/xds.md

```diff
@@ -36,6 +29,113 @@ Since the range of resources configurable by the xDS API extends that of Quilkin
   * Only the list of [filters][xds-filters] specified in the [filter chain][xds-filter-chain] is used by the proxy - i.e other fields like `filter_chain_match` are ignored. This list also specifies the order that the corresponding filter chain will be constructed.
   * gRPC proto configuration for Quilkin's built-in filters [can be found here][filter-protos]. They are equivalent to the filter's static configuration.
 
+## Available Providers
+The server can be run by a quilkin commmand name _manage_.
+
+ ### Agones
+
+1. Cluster information is retrieved from [Agones] - the server watches for `Allocated`
+   [Agones GameServers] and exposes their IP address and Port as [upstream endpoints][upstream-endpoint] to
+   any connected Quilkin proxies.
+   The set of tokens for the associated endpoint can be set by adding a comma separated standard base64 encoded strings.
+   This must be added under an annotation `quilkin.dev/tokens` in the [GameServer][Agones GameServers]'s spec.
+   For example:
+   ```yaml
+   annotations:
+     Sets two tokens for the corresponding endpoint with values 1x7ijy6 and 8gj3v2i respectively.
+     quilkin.dev/tokens: MXg3aWp5Ng==,OGdqM3YyaQ==
+   ```
+
+   > Since an Agones GameServer can have multiple ports exposed, if multiple ports are in
+   > use, the server looks for the port named `default` and picks that as the endpoint's
+   > port (otherwise it picks the first port in the port list).
+
+2. Filter chain is configurable on a per-proxy basis. By default an empty filter chain is used and from there the filter chain can configured using a configMap name `quilkin-config` on the proxy's pod.
+
+As an example, the following runs the server with subcommnad _agones_ against a cluster (using default kubeconfig configuration) where Quilkin pods run in the `quilkin` namespace and game-server pods run in the `gameservers` namespace:
+
+```sh
+quilkin manage --port 18000 agones --config-namespace quilkin --gameservers-namespace gameservers
+```
+
+> A proxy's pod must have a `quilkin.dev/role` key in `quilkin-config` configMap set to the value `proxy` in order for the management server to detect the pod as a proxy and push updates to it.
+
+> Note that currently, the server can only discover resources within a single cluster.
+
+### Filesystem
+
+The file server command is primarily an example and mostly suitable for demo purposes. As a result, some configuration options and features might be missing. This file implementation watches a configuration file on disk and sends updates to proxies whenever that file changes.
```

Comment:
I think the comment about it being demo-only is not needed, you totally could use it in production, there'd be nothing wrong with it for deployments that are mostly static and/or small.

```suggestion
The filesystem provider watches a configuration file on disk and sends updates to proxies whenever that file changes.
```